tests/integration/MediaWiki/Api/EditEndpointTest.php

```diff
@@ -400,6 +400,44 @@ public function testReconciliationPropertyMissingInStatements(): void {
 			$exception->getMessageValue()->getKey() );
 	}
 
+	public function testInvalidSnakTypeValue(): void {
+		/** @var LocalizedHttpException $exception */
+		$exception = $this->executeHandlerAndGetHttpException(
+			$this->newHandler(),
+			$this->newRequest( [
+				'entity' => [
+					EditEndpoint::VERSION_KEY => '0.0.1/full',
+					'type' => 'item',
+					'claims' => [
+						self::URL_PROPERTY => [
+							[
+								'mainsnak' => [
+									'snaktype' => 'somevalue',
+									'property' => self::URL_PROPERTY,
+									'datavalue' => [
```

Comment:
somevalue snaks should not have a datavalue.

Reply:
Done. thanks I didn't know this.